apis/notebooks/v1beta1/instance_types.go

```diff
@@ -0,0 +1,292 @@
+	// +kcc:proto:field=google.cloud.notebooks.v1.Instance.upgrade_history
+	UpgradeHistory []Instance_UpgradeHistoryEntry `json:"upgradeHistory,omitempty"`
+
+	// Optional. The type of vNIC to be used on this interface. This may be gVNIC or
+	//  VirtioNet.
+	// +kcc:proto:field=google.cloud.notebooks.v1.Instance.nic_type
+	NicType *string `json:"nicType,omitempty"`
+
+	// Optional. The optional reservation affinity. Setting this field will apply
+	//  the specified [Zonal Compute
+	//  Reservation](https://cloud.google.com/compute/docs/instances/reserving-zonal-resources)
+	//  to this notebook instance.
+	// +kcc:proto:field=google.cloud.notebooks.v1.Instance.reservation_affinity
+	ReservationAffinity *ReservationAffinity `json:"reservationAffinity,omitempty"`
+
+	// Optional. Flag to enable ip forwarding or not, default false/off.
+	//  https://cloud.google.com/vpc/docs/using-routes#canipforward
+	// +kcc:proto:field=google.cloud.notebooks.v1.Instance.can_ip_forward
+	CanIPForward *bool `json:"canIPForward,omitempty"`
+
+	// Required. The parent resource name where the Job will be created. Pattern: "projects/{project}/locations/{location}"
+	*Parent `json:",inline"`
+
+	// The NotebookInstance name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+}
+
+type Parent struct {
+	// Immutable. The location where the notebook instance should reside.
+	// +required
+	Zone string `json:"zone,omitempty"`
+
+	// The project that this resource belongs to.
+	// +required
+	ProjectRef *v1beta1.ProjectRef `json:"projectRef,omitempty"`
+}
+
+// NotebookInstanceStatus defines the config connector machine state of NotebookInstance
+type NotebookInstanceStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the NotebookInstance resource in GCP.
+	ExternalRef *string `json:"externalRef,omitempty"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *NotebookInstanceObservedState `json:"observedState,omitempty"`
+}
+
+// NotebookInstanceObservedState is the state of the NotebookInstance resource as most recently observed in GCP.
+// +kcc:proto=google.cloud.notebooks.v1.Instance
+type NotebookInstanceObservedState struct {
+	// Output only. The name of this notebook instance. Format:
+	//  `projects/{project_id}/locations/{location}/instances/{instance_id}`
+	// +kcc:proto:field=google.cloud.notebooks.v1.Instance.name
+	Name *string `json:"name,omitempty"`
```

Comment:
Maybe I missed in the alpha review. Name field seems identical to the externalRef? (in which case it should be removed)